homeassistant/components/netgear/router.py

```diff
@@ -204,6 +204,13 @@ async def async_get_traffic_meter(self) -> dict[str, Any] | None:
         async with self._api_lock:
             return await self.hass.async_add_executor_job(self._api.get_traffic_meter)
 
+    async def async_get_speed_test(self) -> dict[str, Any] | None:
+        """Perform a speed test and get the results from the router."""
+        async with self._api_lock:
+            return await self.hass.async_add_executor_job(
+                self._api.get_new_speed_test_result
```

Comment:
This can potentially block the lock for 20 seconds.

https://github.com/MatMaul/pynetgear/blob/9fa293a6fa0a639e83fc54ddd3abba94acc21a0f/pynetgear/router.py#L931-L952

Generally, we don't want integrations to sleep in our executor thread pool, as it may exhaust the pool.

Specifically, this may stagger other api calls that come in during the time period that the lock blocks. The mitigating factor is only the much longer time interval for the speed test calls. That stops this from becoming a problem that breaks the integration. The design is still bad though.